accessibility/common-remediations.md

```diff
@@ -0,0 +1,197 @@
+<button> Be focused<button />
+<button> Last <button />
+```
+
+We don't do this:
+```html
+<button style=”float: right”> I Should <button />
+<button> Be focused<button />
+<button> Last <button />
+```
+
+
+
+### No advance warning when opening new window 
+
+Opening new windows automatically when a link is activated can be disorienting for people who have difficulty perceiving visual content, and for some people with cognitive disabilities, if they are not warned in advance.
+
+We do this:
+```html
+<a href="https://mysite.com" target="_blank" rel="noopener">
+ My site
+ <span class="visuallyhidden">, opens in a new window</span>
+</a>
+```
+
+We don’t do this:
+```html
+<a href="knitting.html" target="_blank">All about Knitting</a>
+```
+
+
+
+## Images
+
+### Images with no alt attribute
+
+Alt attributes are required on all `img` tags. Your HTML is invalid without them. 
+
+We do this:
+```html
+<img src="/images/NYCpark.png" alt="Aerial view of Central Park in New York" />
+```
+
+We don't do this:
+```html
+<img src="/images/NYCpark.png" />
+```
+
+
+### Meaningful images with meaningless alt attributes
+
+Alt text should be used to convey the intent of an image. Avoid using generic strings like “photo”, “image”, or “icon” as alt values, as they don’t communicate valuable content to the user. Be as descriptive as possible.
+
+We do this:
+```html
+<img src="/images/kinghenry.jpg" alt="King Henry VIII of England" />
+```
+
+We don't do this:
+```html
+ <img src="/images/kinghenry.jpg" alt="Image of King" />
+```
+
+
+
+### Meaningless images with meaningful alt attributes
+
+If an image is decorative, the alt attribute should be present, but its value left blank. This conveys that the image isn’t necessary for understanding the page.
+
+We do this:
+```html
+ <img src="/images/spacer.gif" alt="" />
+ <img src="/images/bg_img.gif" alt="" />
+```
+
+We _don’t_ do this:
+```html
+ <img src="/images/spacer.gif" alt="spacer image" />
+ <img src="/images/bg_img.gif" alt="background image" />
+```
+
+
+
+### Redundant alt attributes
+If an image represents information that is already available in text, the alt attribute should be present, but its value left blank. This prevents screen reader users from having to listen to multiple repetitions of the same information. 
+
+We do this:
+```html
+<div class=”person”><img src="/images/mrs-a-person.gif" alt=""><a href=”/”>Mrs. A. Person</a></div>
+```
+
+We _don’t_ do this:
+```html
+<div class=”person”><img src="/images/mrs-a-person.gif" alt="Mrs. A. Person" /><a href=”/”>Mrs. A. Person</a></div>
+```
+
+
+
+### Logo image
+
+For an image `alt` attribute is sufficient and more meaningful.
+No need of title attribute, use an inline image with an alt attribute for the brand logo.
+Make sure any text in images of text is at least 14 points and has good contrast with the background.
+
+We do this:
+```html
+<a href="https://www.w3.org/"> <img src="w3c.png" alt="W3C home"></a>
```

Comment:
```suggestion
<a href="https://www.w3.org/"> <img src="w3c.png" alt="W3C"></a>
```

Alt text must always match the text in the image exactly - no words should be added before or after.  We need a separate heading for this rule, too!